base/sim/FairMCApplication.cxx

```diff
@@ -599,13 +602,14 @@ void FairMCApplication::PreTrack()
 //_____________________________________________________________________________
 TVirtualMCApplication* FairMCApplication::CloneForWorker() const
 {
+  mtx.lock();
```

Comment:
```suggestion
  std::unique_lock<std::mutex> lock(mtx);
```
Use this and remove mtx.unlock(); at the end, unique_lock will do lock/unlock for you in its constructor/destructor. This will make the locking exception-safe.